code/modules/mob/mob_misc_procs.dm

```diff
@@ -560,7 +560,7 @@
 						lname = name
 				lname = "<span class='name'>[lname]</span> "
 			to_chat(M, "<span class='deadsay'>[lname][follow][message]</span>")
-			if(should_show_runechat && (M.client?.prefs.toggles2 & PREFTOGGLE_2_RUNECHAT) && M.see_invisible >= M.invisibility)
+			if(should_show_runechat && (M.client?.prefs.toggles2 & PREFTOGGLE_2_RUNECHAT) && (M in viewers("19x15", subject)))
```

Comment:
God please just override the plane on spawn with an arg on `create_chat_message`.

Viewers is costlyyyyyyyyyyyy